File: bm_front/src/Network/Dto/Banner/UpdateBannerDto.ts
export class UpdateBannerDto {
    id: number
    name: string
    price: number
    categoryId: number
    text: string

    constructor(id: number, name: string = "", price: number = 0, categoryId: number = -1, text: string = "") {
        this.id = id;
        this.name = name;
        this.price = price;
        this.categoryId = categoryId;
        this.text = text;
    }
}

File: banner_manager/src/main/java/ru/nsu/fit/borzov/banner_manager/dto/UpdateBannerDto.java
package ru.nsu.fit.borzov.banner_manager.dto;

import lombok.Data;

import javax.validation.constraints.NotNull;

@Data
public class UpdateBannerDto {

    @NotNull
    private Long id;

    @NotNull
    private String name;

    @NotNull
    private Double price;

    @NotNull
    private String text;

    @NotNull
    private Long categoryId;
}
